### PR TITLE
Combine SonarQube analysis build with the main build on Maven.

### PR DESCRIPTION
**Pull request by HSAR**

### CODE REVIEW

File: definitions/sonarqube-common.d.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ declare module 'sonarqube-common/sonarqube-common' {
     export function applySonarQubeAnalysisParams(toolRunner:ToolRunner, projectName?, projectKey?, projectVersion?):ToolRunner;
 
     // Fetches configured SonarQube endpoint details.
-    export function getSonarQubeEndpointFromInput(inputFieldName):SonarQubeEndpoint;
+    export function getSonarQubeEndpoint():SonarQubeEndpoint;
```
Comment (HSAR) [inline definitions/sonarqube-common.d.ts]: Reducing arguments to gather them up inside sonarqube-common rather than inside the individual sonar__.ts files.

File: definitions/sonarqube-common.d.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ declare module 'sonarqube-common/sonarqube-common' {
 
     // Applies required parameters for connecting a Java-based plugin (Maven, Gradle) to SonarQube.
     // sqDbUrl, sqDbUsername and sqDbPassword are required if the SonarQube version is less than 5.2.
-    export function applySonarQubeConnectionParams(toolRunner:ToolRunner, sqHostUrl, sqHostUsername, sqHostPassword, sqDbUrl?, sqDbUsername?, sqDbPassword?):ToolRunner;
+    export function applySonarQubeConnectionParams(toolRunner:ToolRunner): ToolRunner;
```
Comment (HSAR) [inline definitions/sonarqube-common.d.ts]: We may as well do the getInput calls inside sonarqube-common rather than inside each individual __sonar.ts files.

File: Tasks/Common/sonarqube-common/sonarqube-common.ts [hunk 3]
```diff
@@ -196,6 +201,18 @@ function getOrCreateSonarQubeStagingDirectory(): string {
     return sqStagingDir;
 }
 
+// Creates a string containing Markdown of a link to the SonarQube dashboard for this project.
```
Comment (HSAR) [inline Tasks/Common/sonarqube-common/sonarqube-common.ts]: Unrelated change: tidying up the build summary code in preparation for the next ticket.

File: Tasks/Common/sonarqube-common/sonarqube-common.ts [hunk 4]
```diff
@@ -23,23 +23,29 @@ export function isSonarQubeAnalysisEnabled(): boolean {
 }
 
 // Applies required parameters for connecting a Java-based plugin (Maven, Gradle) to SonarQube.
-// sqDbUrl, sqDbUsername and sqDbPassword are required if the SonarQube version is less than 5.2.
-export function applySonarQubeConnectionParams(toolRunner: ToolRunner, sqHostUrl, sqHostUsername, sqHostPassword, sqDbUrl?, sqDbUsername?, sqDbPassword?): ToolRunner {
-    toolRunner.arg('-Dsonar.host.url=' + sqHostUrl);
-    toolRunner.arg('-Dsonar.login=' + sqHostUsername);
-    toolRunner.arg('-Dsonar.password=' + sqHostPassword);
+export function applySonarQubeConnectionParams(mvnRun: ToolRunner): ToolRunner {
```
Comment (bgavrilMS) [inline Tasks/Common/sonarqube-common/sonarqube-common.ts]: If this is common code, why is the generic tool runner called mvnRun ?

File: Tasks/Gradle/CodeAnalysis/gradlesonar.ts [hunk 5]
```diff
@@ -28,25 +28,17 @@ export function applyEnabledSonarQubeArguments(gradleRun: trm.ToolRunner):trm.To
...
+
+    gradleRun = sqCommon.applySonarQubeConnectionParams(gradleRun);
 
     // Add parameters to specify the SonarQube project properties (if given by the user)
     var projectName:string = tl.getInput('sqProjectName', true);
     var projectKey:string = tl.getInput('sqProjectKey', true);
     var projectVersion:string = tl.getInput('sqProjectVersion', true);
-    
     gradleRun = sqCommon.applySonarQubeAnalysisParams(gradleRun, projectName, projectKey, projectVersion);
+
```
Comment (bgavrilMS) [inline Tasks/Gradle/CodeAnalysis/gradlesonar.ts]: You can encapsulate these 3 methods inside a single common one that takes projectName, key and version as optional params.

File: Tasks/Common/sonarqube-common/sonarqube-common.ts (hunk 4 above)
Comment (HSAR) [inline Tasks/Common/sonarqube-common/sonarqube-common.ts]: Accepted: changing back to toolRunner.

Comment (HSAR) [inline Tasks/Common/sonarqube-common/sonarqube-common.ts]: Resolved.

File: Tasks/Maven/maventask.ts [hunk 6]
```diff
@@ -129,8 +128,7 @@ else {
 // 7. If #2, #4 or #5 above failed, exit with an error code to mark the entire step as failed.
 
 var userRunFailed: boolean = false;
-var sonarQubeRunFailed: boolean = false;
-var pmdRunFailed: boolean = false;
+var codeAnalysisFailed: boolean = false;
```
Comment (bgavrilMS) [inline Tasks/Maven/maventask.ts]: There are some comments related to orchestration that need to be updated.

**Comment by bgavrilMS** [comment]
LGTM

File: Tasks/Gradle/CodeAnalysis/gradlesonar.ts (hunk 5 above)
Comment (HSAR) [inline Tasks/Gradle/CodeAnalysis/gradlesonar.ts]: Resolved, done.

File: Tasks/Maven/maventask.ts (hunk 6 above)
Comment (HSAR) [inline Tasks/Maven/maventask.ts]: Accepted.

Comment (HSAR) [inline Tasks/Maven/maventask.ts]: Resolved.

Comment (HSAR) [inline Tasks/Maven/maventask.ts]: Resolved.
